### ISSUE TITLE
Quick switch to next video program will stuck and no response

### ISSUE THREAD

**Issue by kingeaglewang** [higankanshi/Meta.Vlc#166]
Hi, I use the code from the git.
After user loads a video, I want to let user click next button to simply switch to next video in the same folder.

I create a next button,  and it can show next video.
The  click function of next button is like following.

private void NextButton_Click(object sender, RoutedEventArgs e)
{
            if (nowIndex == maxVideoNumber)
                nowIndex = 0;
            else
                nowIndex++;
            Player.Stop();
            Player.LoadMedia(FileName[nowIndex]);
            Player.Play();
 }

If I click the next button very quickly, one of the two problems will occur.

1. The player will stuck and no response. But it won't show a dialog say the program is crash.  It just stuck I can't close it normally. I can only close it by task manager.
2. The player will only have audio but without video.

If I click pause in Visual studio, It stops at the following function.
  public void Stop()
        {
            _stopFunction.Delegate(InstancePointer);
        }

Is any idea about these problems ?
Thanks 

p.s:  I try on the real VLC player.  If I click next video very quickly, it won't get stuck.

**Comment by higankanshi** [comment]
Did you  have tried the Github version of Meta.Vlc.
I have do many changes to make switch video work fine.

I will update the Nuget in this month.

**Comment by kingeaglewang** [comment]
Yes, I try the github version.
You need to switch video very quickly. If you use load function to switch video, it is too slow.
So, I create a button, when I click the button, it will switch to next video.

If I click the button very quickly, the program will get stuck.

**Comment by mahdi-ataollahi** [comment]
I have the same issue. I have some online video files. When I download them and play them from local disk, at the first time it plays but when I repeat it or I try play it later (after playing another video) it doesn't have video and it only plays audio.
I used this code:

VideoControl.Stop();
VideoControl.RebuildPlayer();
VideoControl.LoadMedia(url);
VideoControl.Play();

If I set 'url' as something like "http://ip:port/.../video.mp4" it works perfect every time I play but when I use @"D:\...\video.mp4" it works at first time but not in next plays.
